tests/flow/test_unwind_clause.py

```diff
@@ -0,0 +1,88 @@
+from common import *
+import re
+
+redis_graph = None
+
+class testUnwindClause(FlowTestsBase):
+    
+    def __init__(self):
+        self.env = Env(decodeResponses=True)
+        global redis_graph
+        redis_con = self.env.getConnection()
+        redis_graph = Graph(redis_con, "G")
+ 
+    def test01_unwind_null_output(self):
+        query = """UNWIND ({x:null}) AS q RETURN q.x"""
+        actual_result = redis_graph.query(query)
+        expected = [[None]]
+        self.env.assertEqual(actual_result.result_set, expected)
+
+        query = """UNWIND null AS x RETURN x"""
+        actual_result = redis_graph.query(query)
+        expected = [[None]]
+        self.env.assertEqual(actual_result.result_set, expected)
+
+    def test02_unwind_input_types(self):
+        # map list input
+        query = """UNWIND ([{x:3, y:5}]) AS q RETURN q"""
+        actual_result = redis_graph.query(query)
+        expected = [[{'x':3, 'y':5}]]
+        self.env.assertEqual(actual_result.result_set, expected)
+
+        # map input
+        query = """UNWIND ({x:3, y:5}) AS q RETURN q"""
+        actual_result = redis_graph.query(query)
+        expected = [[{'x': 3, 'y': 5}]]
+        self.env.assertEqual(actual_result.result_set, expected)
+
+        # integer input
+        query = """UNWIND 5 AS q RETURN q"""
+        actual_result = redis_graph.query(query)
+        expected = [[5]]
+        self.env.assertEqual(actual_result.result_set, expected)
+
+        # string input
+        query = """UNWIND 'abc' AS q RETURN q"""
+        actual_result = redis_graph.query(query)
+        expected = [['abc']]
+        self.env.assertEqual(actual_result.result_set, expected)
+
+        # floating-point input
+        query = """UNWIND 7.5 AS q RETURN q"""
+        actual_result = redis_graph.query(query)
+        expected = [[7.5]]
+        self.env.assertEqual(actual_result.result_set, expected)
+
+        # nested list
+        query = """WITH [[1, 2], [3, 4], 5] AS nested UNWIND nested AS x RETURN x"""
+        actual_result = redis_graph.query(query)
+        expected = [[[1, 2]], [[3, 4]], [5]]
+        self.env.assertEqual(actual_result.result_set, expected)
+
+        # nested list double unwind
+        query = """WITH [[1, 2], [3, 4], 5] AS nested UNWIND nested AS x UNWIND x AS y RETURN y"""
+        actual_result = redis_graph.query(query)
+        expected = [[1], [2], [3], [4], [5]]
+        self.env.assertEqual(actual_result.result_set, expected)
+
+        # empty list
+        query = """UNWIND [] AS x RETURN x"""
+        actual_result = redis_graph.query(query)
+        expected = []
+        self.env.assertEqual(actual_result.result_set, expected)
+
+        # list including null
+        query = """UNWIND [1, 2, null] AS x RETURN x"""
+        actual_result = redis_graph.query(query)
+        expected = [[1], [2], [None]]
+        self.env.assertEqual(actual_result.result_set, expected)
```

Comment:
Please add `UNWIND [1, null, 2] AS x RETURN x`
I think the iteration should halt after `NULL` and `2` won't be omitted? please double check this against Neo4J

We might want to make sure this behavior is covered by TCK

Reply:
The iteration does not halt after ```NULL```, the tests were done in Neo4J.

I think that this behavior is covered by TCK test like the test in TypeConversion1 [Scenario: [4] `toBoolean()` on invalid strings](https://github.com/RedisGraph/RedisGraph/blob/ad07715bfb8c27f604bc6b2a1fb4a4bc55ac8238/tests/tck/features/expressions/typeConversion/TypeConversion1.feature#L70)
where the input list has a null at first position.